plugins/cairo-lang-macro-attributes/src/lib.rs

```diff
@@ -28,3 +29,40 @@ pub fn attribute_macro(_args: TokenStream, input: TokenStream) -> TokenStream {
     };
     TokenStream::from(expanded)
 }
+
+/// This macro can be used to construct the auxiliary data collection callback.
+///
+/// The procedural macro can emit additional auxiliary data alongside the generated [`TokenStream`]
+/// during the code expansion. This data can be used to collect additional information from the
+/// source code of a project that is being compiled during the macro execution.
+/// For instance, you can create a procedural macro that collects some information stored by
+/// the Cairo programmer as attributes in the project source code.
+///
+/// This should be used to implement a collection callback for the auxiliary data.
+/// This callback will be called after the source code compilation (and thus after all the procedural
+/// macro executions). All auxiliary data emitted by the procedural macro during source code compilation
+/// will be passed to the callback as an argument.
+///
+/// The callback can be used to process or persist the data collected during the compilation.
+///
+/// This macro hides the conversion to stable ABI structs from the user.
+///
+/// # Safety
+/// Note that AuxData deserialization may fail.
```

Comment:
what happens when it fails? do we panic? ignore? abort?

Reply:
I do not think it's still relevant since we do not rely on serde in AuxData internals no more.